feat(traverse): implement `GatherNodeParts` for member expression types

overlookmotel opened the pull request:
`GatherNodeParts` was missing support for these types as entry points for collection.

overlookmotel commented:
* **#7011** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7011?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7122** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7122?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7363** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7363?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7363?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
### Merge activity

* **Nov 19, 10:58 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
feat(traverse): implement `GatherNodeParts` for member expression types (#7363)

`GatherNodeParts` was missing support for these types as entry points for collection.
